### ISSUE TITLE
akka-http failed with "javax.net.ssl.SSLException: Received fatal alert: handshake_failure" where nio works

### ISSUE THREAD

**Issue by TimothyKlim** [akka/akka#20110]
Hello hAkkers, I'm trying to get file by akka-http (https://github.com/TimothyKlim/akka-http-tls-poc/blob/master/src/main/scala/Main.scala):
```scala
Http()
    .singleRequest(HttpRequest(uri = src))
    .flatMap(_.entity.dataBytes.runFold(0L)(_ + _.length))
    .onComplete(res => println(s"akka-http: $res"))
```
and I always get this strange TLS error: `Aborting tcp connection because of upstream failure: Inbound closed before receiving peer's close_notify: possible truncation attack?` ([full log](https://gist.github.com/TimothyKlim/d58f8b438ed91c4f862c)).
I can successfully download "problem" file through java nio or even curl:
```scala
val url = new URL(src)
val rbc = Channels.newChannel(url.openStream())
val fos = new FileOutputStream(Files.createTempFile(null, null).toFile)
```

POC project: https://github.com/TimothyKlim/akka-http-tls-poc

**Comment by ktoso** [comment]
The sample url causing the trouble is `https://ucarecdn.com/2877682a-c715-4848-8fe2-4ff9fbd7728e/`

**Comment by johanandren** [comment]
Works in 2.4.3, I'm guessing it was fixed by the fixes for #19287 and #16796
